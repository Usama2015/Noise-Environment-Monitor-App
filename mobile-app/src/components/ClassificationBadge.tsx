/**
 * ClassificationBadge Component
 *
 * Displays the noise classification category with color-coded badge,
 * icon, and descriptive text.
 *
 * Features:
 * - Color-coded badge (green/yellow/red)
 * - Category icon (emoji)
 * - Category label
 * - Optional description
 * - Confidence indicator
 *
 * @see PROJECT_PLAN.md - Step 1.7: Basic UI Implementation
 */

import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import type { NoiseCategory } from '../services/NoiseClassifier';

export interface ClassificationBadgeProps {
  /** Noise category */
  category: NoiseCategory;
  /** Hex color code for badge */
  color: string;
  /** Icon for category (emoji) */
  icon: string;
  /** Optional description of noise type */
  description?: string;
  /** Optional confidence score (0-1) */
  confidence?: number;
  /** Optional detailed category info */
  categoryInfo?: string;
}

/**
 * ClassificationBadge component
 *
 * @example
 * ```tsx
 * <ClassificationBadge
 *   category="Normal"
 *   color="#FFC107"
 *   icon="ðŸŸ¡"
 *   description="Voice / Music"
 *   confidence={0.85}
 * />
 * ```
 */
export const ClassificationBadge: React.FC<ClassificationBadgeProps> = ({
  category,
  color,
  icon,
  description,
  confidence,
  categoryInfo,
}) => {
  return (
    <View style={styles.container}>
      {/* Main badge */}
      <View style={[styles.badge, { backgroundColor: color }]}>
        <Text style={styles.icon}>{icon}</Text>
        <Text style={styles.categoryText}>{category}</Text>
      </View>

      {/* Noise type description */}
      {description && (
        <Text style={styles.description}>{description}</Text>
      )}

      {/* Confidence indicator */}
      {confidence !== undefined && (
        <View style={styles.confidenceContainer}>
          <Text style={styles.confidenceLabel}>Confidence:</Text>
          <View style={styles.confidenceBar}>
            <View
              style={[
                styles.confidenceFill,
                {
                  width: `${confidence * 100}%`,
                  backgroundColor: getConfidenceColor(confidence),
                },
              ]}
            />
          </View>
          <Text style={styles.confidenceValue}>
            {(confidence * 100).toFixed(0)}%
          </Text>
        </View>
      )}

      {/* Category information */}
      {categoryInfo && (
        <View style={styles.infoContainer}>
          <Text style={styles.infoText}>{categoryInfo}</Text>
        </View>
      )}
    </View>
  );
};

/**
 * Get color for confidence bar based on confidence level
 */
function getConfidenceColor(confidence: number): string {
  if (confidence >= 0.8) return '#4CAF50'; // High confidence - green
  if (confidence >= 0.6) return '#FFC107'; // Medium confidence - yellow
  return '#FF9800'; // Low confidence - orange
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginVertical: 16,
  },
  badge: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 12,
    borderRadius: 25,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.15,
    shadowRadius: 4,
    elevation: 3,
  },
  icon: {
    fontSize: 24,
    marginRight: 8,
  },
  categoryText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#fff',
  },
  description: {
    fontSize: 16,
    color: '#666',
    marginTop: 12,
    fontStyle: 'italic',
  },
  confidenceContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 12,
    width: '80%',
  },
  confidenceLabel: {
    fontSize: 12,
    color: '#999',
    marginRight: 8,
    width: 70,
  },
  confidenceBar: {
    flex: 1,
    height: 8,
    backgroundColor: '#E0E0E0',
    borderRadius: 4,
    overflow: 'hidden',
  },
  confidenceFill: {
    height: '100%',
    borderRadius: 4,
  },
  confidenceValue: {
    fontSize: 12,
    fontWeight: '600',
    color: '#666',
    marginLeft: 8,
    width: 40,
    textAlign: 'right',
  },
  infoContainer: {
    marginTop: 12,
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: '#F5F5F5',
    borderRadius: 8,
    maxWidth: '90%',
  },
  infoText: {
    fontSize: 12,
    color: '#666',
    textAlign: 'center',
    lineHeight: 16,
  },
});
