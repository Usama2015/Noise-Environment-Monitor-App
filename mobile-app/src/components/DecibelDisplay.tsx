/**
 * DecibelDisplay Component
 *
 * Displays the current decibel level with a large, animated number
 * and visual color indicator based on noise category.
 *
 * Features:
 * - Large, prominent decibel value
 * - Smooth number transitions
 * - Color-coded background based on category
 * - Responsive sizing
 *
 * @see PROJECT_PLAN.md - Step 1.7: Basic UI Implementation
 */

import React, { useEffect, useRef } from 'react';
import { Animated, StyleSheet, Text, View } from 'react-native';
import type { NoiseCategory } from '../services/NoiseClassifier';

export interface DecibelDisplayProps {
  /** Current decibel value */
  decibels: number;
  /** Noise category for color coding */
  category: NoiseCategory;
  /** Optional label */
  label?: string;
}

/**
 * Get background color based on noise category
 */
function getCategoryBackgroundColor(category: NoiseCategory): string {
  switch (category) {
    case 'Quiet':
      return '#E8F5E9'; // Light green
    case 'Normal':
      return '#FFF9C4'; // Light yellow
    case 'Noisy':
      return '#FFEBEE'; // Light red
  }
}

/**
 * Get text color based on noise category
 */
function getCategoryTextColor(category: NoiseCategory): string {
  switch (category) {
    case 'Quiet':
      return '#2E7D32'; // Dark green
    case 'Normal':
      return '#F57C00'; // Dark orange
    case 'Noisy':
      return '#C62828'; // Dark red
  }
}

/**
 * DecibelDisplay component
 *
 * @example
 * ```tsx
 * <DecibelDisplay
 *   decibels={65.4}
 *   category="Normal"
 *   label="Current Level"
 * />
 * ```
 */
export const DecibelDisplay: React.FC<DecibelDisplayProps> = ({
  decibels,
  category,
  label = 'Sound Level',
}) => {
  // Animated value for smooth transitions
  const fadeAnim = useRef(new Animated.Value(1)).current;
  const scaleAnim = useRef(new Animated.Value(1)).current;

  // Pulse animation when decibel value changes significantly
  useEffect(() => {
    // Pulse effect
    Animated.sequence([
      Animated.parallel([
        Animated.timing(scaleAnim, {
          toValue: 1.05,
          duration: 100,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 0.8,
          duration: 100,
          useNativeDriver: true,
        }),
      ]),
      Animated.parallel([
        Animated.timing(scaleAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        }),
      ]),
    ]).start();
  }, [decibels, category, fadeAnim, scaleAnim]);

  const backgroundColor = getCategoryBackgroundColor(category);
  const textColor = getCategoryTextColor(category);

  return (
    <View style={styles.container}>
      {label && <Text style={styles.label}>{label}</Text>}

      <Animated.View
        style={[
          styles.displayBox,
          {
            backgroundColor,
            opacity: fadeAnim,
            transform: [{ scale: scaleAnim }],
          },
        ]}
      >
        <Text style={[styles.decibelValue, { color: textColor }]}>
          {decibels.toFixed(1)}
        </Text>
        <Text style={[styles.unit, { color: textColor }]}>dB</Text>
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginVertical: 20,
  },
  label: {
    fontSize: 14,
    fontWeight: '500',
    color: '#666',
    marginBottom: 8,
    textTransform: 'uppercase',
    letterSpacing: 1,
  },
  displayBox: {
    paddingHorizontal: 40,
    paddingVertical: 30,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    minWidth: 200,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
  decibelValue: {
    fontSize: 64,
    fontWeight: 'bold',
    lineHeight: 72,
  },
  unit: {
    fontSize: 24,
    fontWeight: '600',
    marginTop: 4,
  },
});
