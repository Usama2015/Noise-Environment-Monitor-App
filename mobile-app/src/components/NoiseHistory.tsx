/**
 * NoiseHistory Component
 *
 * Displays a list of recent noise readings with decibel values
 * and classifications.
 *
 * Features:
 * - Scrollable list of readings
 * - Color-coded entries
 * - Timestamp display
 * - Limit to last N readings
 *
 * @see PROJECT_PLAN.md - Step 1.7: Basic UI Implementation
 */

import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import type { NoiseCategory } from '../services/NoiseClassifier';

export interface NoiseReading {
  /** Decibel value */
  decibels: number;
  /** Classification category */
  category: NoiseCategory;
  /** Timestamp */
  timestamp: Date;
  /** Optional description */
  description?: string;
}

export interface NoiseHistoryProps {
  /** Array of noise readings */
  readings: NoiseReading[];
  /** Maximum number of readings to display */
  maxReadings?: number;
  /** Show timestamps */
  showTimestamps?: boolean;
}

/**
 * Get color dot for category
 */
function getCategoryDot(category: NoiseCategory): string {
  switch (category) {
    case 'Quiet':
      return 'ðŸŸ¢';
    case 'Normal':
      return 'ðŸŸ¡';
    case 'Noisy':
      return 'ðŸ”´';
  }
}

/**
 * Format timestamp for display
 */
function formatTimestamp(date: Date): string {
  const now = new Date();
  const diffMs = now.getTime() - date.getTime();
  const diffSecs = Math.floor(diffMs / 1000);

  if (diffSecs < 60) {
    return `${diffSecs}s ago`;
  }

  const diffMins = Math.floor(diffSecs / 60);
  if (diffMins < 60) {
    return `${diffMins}m ago`;
  }

  const diffHours = Math.floor(diffMins / 60);
  if (diffHours < 24) {
    return `${diffHours}h ago`;
  }

  return date.toLocaleTimeString();
}

/**
 * NoiseHistory component
 *
 * @example
 * ```tsx
 * <NoiseHistory
 *   readings={recentReadings}
 *   maxReadings={10}
 *   showTimestamps={true}
 * />
 * ```
 */
export const NoiseHistory: React.FC<NoiseHistoryProps> = ({
  readings,
  maxReadings = 10,
  showTimestamps = true,
}) => {
  // Limit to max readings and sort by timestamp (newest first)
  const displayReadings = readings
    .slice(-maxReadings)
    .reverse();

  if (displayReadings.length === 0) {
    return (
      <View style={styles.emptyContainer}>
        <Text style={styles.emptyText}>No readings yet</Text>
        <Text style={styles.emptySubtext}>
          Start monitoring to see history
        </Text>
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Recent Readings</Text>

      <ScrollView
        style={styles.scrollView}
        contentContainerStyle={styles.scrollContent}
        showsVerticalScrollIndicator={false}
      >
        {displayReadings.map((reading, index) => (
          <View
            key={`${reading.timestamp.getTime()}-${index}`}
            style={styles.readingItem}
          >
            <View style={styles.readingLeft}>
              <Text style={styles.dot}>{getCategoryDot(reading.category)}</Text>
              <View style={styles.readingInfo}>
                <View style={styles.readingMain}>
                  <Text style={styles.decibelText}>
                    {reading.decibels.toFixed(1)} dB
                  </Text>
                  <Text style={styles.categoryText}>{reading.category}</Text>
                </View>
                {reading.description && (
                  <Text style={styles.descriptionText} numberOfLines={1}>
                    {reading.description}
                  </Text>
                )}
              </View>
            </View>

            {showTimestamps && (
              <Text style={styles.timestamp}>
                {formatTimestamp(reading.timestamp)}
              </Text>
            )}
          </View>
        ))}
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
  title: {
    fontSize: 16,
    fontWeight: '600',
    color: '#333',
    marginBottom: 12,
    paddingHorizontal: 16,
  },
  scrollView: {
    flex: 1,
    maxHeight: 300,
  },
  scrollContent: {
    paddingHorizontal: 16,
  },
  readingItem: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 12,
    paddingHorizontal: 12,
    backgroundColor: '#FAFAFA',
    borderRadius: 8,
    marginBottom: 8,
  },
  readingLeft: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
  },
  dot: {
    fontSize: 16,
    marginRight: 12,
  },
  readingInfo: {
    flex: 1,
  },
  readingMain: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  decibelText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#333',
    marginRight: 8,
  },
  categoryText: {
    fontSize: 14,
    color: '#666',
  },
  descriptionText: {
    fontSize: 11,
    color: '#999',
    fontStyle: 'italic',
    marginTop: 2,
  },
  timestamp: {
    fontSize: 11,
    color: '#999',
    marginLeft: 8,
  },
  emptyContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 40,
  },
  emptyText: {
    fontSize: 16,
    fontWeight: '600',
    color: '#999',
    marginBottom: 4,
  },
  emptySubtext: {
    fontSize: 13,
    color: '#BBB',
  },
});
