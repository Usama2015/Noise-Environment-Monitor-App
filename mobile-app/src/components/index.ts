/**
 * Component exports
 *
 * Central export file for all UI components
 */

export { DecibelDisplay } from './DecibelDisplay';
export type { DecibelDisplayProps } from './DecibelDisplay';

export { ClassificationBadge } from './ClassificationBadge';
export type { ClassificationBadgeProps } from './ClassificationBadge';

export { NoiseHistory } from './NoiseHistory';
export type { NoiseHistoryProps, NoiseReading } from './NoiseHistory';
