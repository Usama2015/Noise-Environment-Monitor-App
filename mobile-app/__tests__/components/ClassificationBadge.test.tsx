/**
 * ClassificationBadge Component Tests
 *
 * Tests for the classification badge component including rendering,
 * confidence indicators, and descriptions.
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import { ClassificationBadge } from '../../src/components/ClassificationBadge';

describe('ClassificationBadge', () => {
  describe('Basic rendering', () => {
    it('should render category and icon', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
        />
      );

      expect(getByText('Normal')).toBeTruthy();
      expect(getByText('游리')).toBeTruthy();
    });

    it('should render with description', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
          description="Voice / Music"
        />
      );

      expect(getByText('Normal')).toBeTruthy();
      expect(getByText('Voice / Music')).toBeTruthy();
    });

    it('should render without description', () => {
      const { queryByText } = render(
        <ClassificationBadge
          category="Quiet"
          color="#4CAF50"
          icon="游릭"
        />
      );

      expect(queryByText('Voice / Music')).toBeNull();
    });
  });

  describe('Confidence indicator', () => {
    it('should render confidence bar and value', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
          confidence={0.85}
        />
      );

      expect(getByText('Confidence:')).toBeTruthy();
      expect(getByText('85%')).toBeTruthy();
    });

    it('should format confidence as percentage', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Quiet"
          color="#4CAF50"
          icon="游릭"
          confidence={0.725}
        />
      );

      expect(getByText('73%')).toBeTruthy(); // Rounds to nearest integer
    });

    it('should handle 100% confidence', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Noisy"
          color="#F44336"
          icon="游댮"
          confidence={1.0}
        />
      );

      expect(getByText('100%')).toBeTruthy();
    });

    it('should handle 0% confidence', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
          confidence={0}
        />
      );

      expect(getByText('0%')).toBeTruthy();
    });

    it('should not render confidence if not provided', () => {
      const { queryByText } = render(
        <ClassificationBadge
          category="Quiet"
          color="#4CAF50"
          icon="游릭"
        />
      );

      expect(queryByText('Confidence:')).toBeNull();
    });
  });

  describe('Category information', () => {
    it('should render category info when provided', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Quiet"
          color="#4CAF50"
          icon="游릭"
          categoryInfo="Library, study room, quiet office (<50 dB)"
        />
      );

      expect(getByText('Library, study room, quiet office (<50 dB)')).toBeTruthy();
    });

    it('should not render category info if not provided', () => {
      const { queryByText } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
        />
      );

      expect(queryByText(/Library/)).toBeNull();
    });
  });

  describe('All categories', () => {
    it('should render Quiet badge correctly', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Quiet"
          color="#4CAF50"
          icon="游릭"
        />
      );

      expect(getByText('Quiet')).toBeTruthy();
      expect(getByText('游릭')).toBeTruthy();
    });

    it('should render Normal badge correctly', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
        />
      );

      expect(getByText('Normal')).toBeTruthy();
      expect(getByText('游리')).toBeTruthy();
    });

    it('should render Noisy badge correctly', () => {
      const { getByText } = render(
        <ClassificationBadge
          category="Noisy"
          color="#F44336"
          icon="游댮"
        />
      );

      expect(getByText('Noisy')).toBeTruthy();
      expect(getByText('游댮')).toBeTruthy();
    });
  });

  describe('Snapshot tests', () => {
    it('should match snapshot with minimal props', () => {
      const { toJSON } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with all props', () => {
      const { toJSON } = render(
        <ClassificationBadge
          category="Normal"
          color="#FFC107"
          icon="游리"
          description="Voice / Music"
          confidence={0.87}
          categoryInfo="Conversation, cafeteria, normal office (50-70 dB)"
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
