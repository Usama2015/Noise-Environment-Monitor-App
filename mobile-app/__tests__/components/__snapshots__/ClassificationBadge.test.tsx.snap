// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassificationBadge Snapshot tests should match snapshot with all props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "marginVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 25,
          "elevation": 3,
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        {
          "backgroundColor": "#FFC107",
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "marginRight": 8,
        }
      }
    >
      ðŸŸ¡
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Normal
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 16,
        "fontStyle": "italic",
        "marginTop": 12,
      }
    }
  >
    Voice / Music
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 12,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
          "marginRight": 8,
          "width": 70,
        }
      }
    >
      Confidence:
    </Text>
    <View
      style={
        {
          "backgroundColor": "#E0E0E0",
          "borderRadius": 4,
          "flex": 1,
          "height": 8,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": "100%",
            },
            {
              "backgroundColor": "#4CAF50",
              "width": "87%",
            },
          ]
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 12,
          "fontWeight": "600",
          "marginLeft": 8,
          "textAlign": "right",
          "width": 40,
        }
      }
    >
      87
      %
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#F5F5F5",
        "borderRadius": 8,
        "marginTop": 12,
        "maxWidth": "90%",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "center",
        }
      }
    >
      Conversation, cafeteria, normal office (50-70 dB)
    </Text>
  </View>
</View>
`;

exports[`ClassificationBadge Snapshot tests should match snapshot with minimal props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "marginVertical": 16,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 25,
          "elevation": 3,
          "flexDirection": "row",
          "paddingHorizontal": 20,
          "paddingVertical": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 4,
        },
        {
          "backgroundColor": "#FFC107",
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "marginRight": 8,
        }
      }
    >
      ðŸŸ¡
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Normal
    </Text>
  </View>
</View>
`;
